libcontainer/container_linux.go

```diff
@@ -359,14 +366,10 @@ func (c *linuxContainer) start(process *Process, isInit bool) error {
 		}
 		c.initProcessStartTime = state.InitProcessStartTime
 
-		if c.config.Hooks != nil {
-			bundle, annotations := utils.Annotations(c.config.Labels)
-			s := configs.HookState{
-				Version:     c.config.Version,
-				ID:          c.id,
-				Pid:         parent.pid(),
-				Bundle:      bundle,
-				Annotations: annotations,
+		if c.config.Hooks != nil && len(c.config.Hooks.Poststart) > 0 {
```

Comment:
The `len` additions is really not necessary, and I think it should be removed from all the patches.

If you want, I can just combine this and #1811 in a new PR that I will rebase and fix so we can get 1.0 sooner.

Reply:
> The `len` additions is really not necessary...

From my commit message and topic post:

>  Add `len` guards to avoid computing the state when `.Hooks` is non-`nil` but the particular phase we're looking at is empty.

Are you saying that state computation is so cheap that it's not worth a `len(c.config.Hooks.Poststart) > 0` guard?

Reply:
Yeah, I don't think the state computation is expensive enough that it makes sense to have likely-to-be-incorrectly-copy-pasted checks.

Reply:
> ... likely-to-be-incorrectly-copy-pasted checks.

Wouldn't we expect unit testing to catch this?  If it's not covered now, I think it should be.  The runtime-tools compliance suite should already cover it.

Reply:
And building the status hits the filesystem for at least [this][1], and probably more.

[1]: https://github.com/opencontainers/runc/blob/4769cdf607b0d1cbd73e5407d9655da7511fe193/libcontainer/process_linux.go#L66

Reply:
So they seem like a slam dunk to me, but if you're still unconvinced when I wake back up, I'll drop the guards ;).

Reply:
`/proc` reading is quite cheap, and it should be noted that it will only be hit if `config.Hooks != nil` which IIRC would require having at least one hook defined... I mean, whatever. My main grip with it is that it looks a bit ugly and I guarantee someone will screw it up in a copy-paste.

Reply:
> `/proc` reading is quite cheap...

Compared to, say hitting a disk.  Certainly not compared to a `len` call.

> My main gripe with it is that it looks a bit ugly and I guarantee someone will screw it up in a copy-paste.

And get that screw-up through CI and review?  I'm sceptical ;).  But whatever, I'm not a maintainer.  Rebased onto master (because I'm touching it anyway) and dropped the `len` guards with 6ff5804 -> e238686.